src/main.rs

```diff
@@ -16,7 +16,7 @@ use crate::scanner::Scanner;
 fn main() -> Result<(), Error> {
     let config = match parser::args().config {
         Some(config_path) => Config::load(&config_path)?,
```

Comment:
@thewizzy what are your thoughts on tweaking `Config::load` to take an optional path (replacing this) and tweaking the logic to look like:

```
If arg provided path:
  use it with out seeing if it exists
else if one of the default paths exist:
   use it
else:
   return Ok(Default::default()) 
```

Then we could do something like:

```rust
let config = Config::load(parser::args().config)?
```

Reply:
Added this recommendation. 

I kept `load_file` to keep the loading logic separate from the where logic.